pkg/controller/podautoscaler/horizontal.go

```diff
@@ -678,7 +690,30 @@ func (a *HorizontalController) recordInitialRecommendation(currentReplicas int32
 	}
 }
 
-func (a *HorizontalController) reconcileAutoscaler(ctx context.Context, hpaShared *autoscalingv2.HorizontalPodAutoscaler, key string) error {
+var (
+	// errSpec is used to determine if the error comes from the spec of HPA object in reconcileAutoscaler.
+	// All such errors should have this error as a root error so that the upstream function can understand they're spec errors.
```

Comment:
```suggestion
	// All such errors should have this error as a root error so that the upstream function can distinguish spec errors from internal errors.
```